src/core/colorspace_utils.js

```diff
@@ -0,0 +1,305 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AlternateCS,
+  CalGrayCS,
+  CalRGBCS,
+  DeviceCmykCS,
+  DeviceGrayCS,
+  DeviceRgbaCS,
+  DeviceRgbCS,
+  IndexedCS,
+  LabCS,
+  PatternCS,
+} from "./colorspace.js";
+import { assert, shadow, warn } from "../shared/util.js";
+import { Dict, Name, Ref, RefSet } from "./primitives.js";
+import { IccColorSpace } from "./icc_colorspace.js";
+import { MissingDataException } from "./core_utils.js";
+
+class ColorSpaceUtils {
+  static #failedICCParsing = null;
```

Comment:
Unfortunately I believe that this, since it's global, might be problematic if one worker is used to load multiple different PDFs. (Which the GENERIC library supports via the `worker` getDocument-option or the `workerPort` in GlobalWorkerOptions).

Could we just remove this, and instead rely on our existing colorSpaceCaches to avoid *most* re-parsing?